setup_utils.py

```diff
@@ -0,0 +1,149 @@
+import ast
+import os
+import re
+import subprocess
+
+
+_R_RC = re.compile("rc(\d+)$")
+
+
+def parse_version(path):
+    with open(path) as fp:
+        return _AssignmentParser().parse(fp.read())["version"]
+
+
+def write_version_py(filename, major, minor, micro, is_released,
+                     previous_version=None):
```

Comment:
Yes. The idea is to compute "build numbers" from some previous tag to avoid them increasing to 4-5 digit numbers. We don't have such a tag yet :)